services/proxy/pkg/middleware/oidc_auth.go

```diff
@@ -168,7 +172,7 @@ func (m OIDCAuthenticator) shouldServe(req *http.Request) bool {
 // Authenticate implements the authenticator interface to authenticate requests via oidc auth.
 func (m *OIDCAuthenticator) Authenticate(r *http.Request) (*http.Request, bool) {
 	// there is no bearer token on the request,
-	if !m.shouldServe(r) || isPublicPath(r.URL.Path) {
+	if !m.shouldServe(r) || (isPublicPath(r.URL.Path) && !isInternal(r.URL.Path, m.RevaGatewaySelector)) {
```

Comment:
Hm, wouldn't it be much easier if we'd just remove the `isPublicPath` here. What is wrong with trying to verify the `Bearer` token on e.g. `/dav/public-files` if it's in the request? That way we wouldn't need the `inInternal` check here.

I think the same is true for the `BasicAuthenticator`.

Reply:
If we remove  the `isPublicPath ` we hit the `m.getClaims(token, r)` for every public link which sounds bad
We can filter out the requests to `public-files` with the `sharetoken`

Reply:
> If we remove the `isPublicPath ` we hit the `m.getClaims(token, r)` for every public link which sounds bad

Why is that bad? Also we can easily skip the call and error out when the `Authorizatiion` header does not carry a bearer token.

Reply:
If you mean to check the `token` after retrieving, it makes sense
```
token := strings.TrimPrefix(r.Header.Get(_headerAuthorization), _bearerPrefix)
if token == "" {
    return nil, false
}
```

Reply:
However, some cases will be broken because the sequence of the authenticators has been changed. the `public` and `oidc` are swapped. The request below will be failed.
```
http://example.com/dav/public-files/
-H Authorization Bearer some_token
-H public-token some_tocken 
``` 
It was passed with the `public_share_auth`
But after we swapped authenticators, it will not reach the `public_share_auth`

Reply:
It means all cases that pass in a `public_share_auth` we should skip in an `oidc`.

Reply:
Already changed to `if !m.shouldServe(r)` Let's see what impact this has.